include/xsimd/arch/xsimd_avx2.hpp

```diff
@@ -173,7 +173,15 @@ namespace xsimd {
                 __m256i tmp2 = _mm256_add_epi64(self, tmp1);
                 __m128i tmp3 = _mm256_extracti128_si256(tmp2, 1);
                 __m128i res = _mm_add_epi64(_mm256_castsi256_si128(tmp2), tmp3);
+#if defined(__x86_64__)
```

Comment:
Copied from SSE2 code
https://github.com/xtensor-stack/xsimd/blob/master/include/xsimd/arch/xsimd_sse2.hpp#L426-L434

Reply:
Error log at https://github.com/apache/arrow/pull/11142/checks?check_run_id=3727229675#step:9:1265

Reply:
Yes, it's an issue only on 32 bits.

Reply:
Feel free to pick this change if it's useful.